### PR TITLE
real owner for nft, collection description, my wallets option for cho…

### PR DESCRIPTION
**Pull request by Viscount88**
…ose receiver

### CODE REVIEW

File: fearless/Modules/NFT/NftCollection/NftCollectionViewController.swift [hunk 1]
```diff
@@ -90,7 +90,8 @@ extension NftCollectionViewController: UICollectionViewDataSource, UICollectionV
         case 0:
             let space: CGFloat = (flowlayout?.sectionInset.left ?? 0.0) + (flowlayout?.sectionInset.right ?? 0.0)
             let width = rootView.collectionView.frame.size.width - space
-            return CGSize(width: width, height: width)
+            let expectedLabelHeight = viewModel?.collectionDescription?.height(withConstrainedWidth: width, font: .p1Paragraph) ?? 0
```
Comment (DRadmir) [inline fearless/Modules/NFT/NftCollection/NftCollectionViewController.swift:93]: please take font from rootView

File: fearless/Modules/NFT/NftDetails/NftDetailsInteractor.swift [hunk 2]
```diff
@@ -3,22 +3,61 @@ import UIKit
 final class NftDetailsInteractor {
     // MARK: - Private properties
 
+    private let nftFetchingService: NFTFetchingServiceProtocol
     private weak var output: NftDetailsInteractorOutput?
     private var nft: NFT?
+    private var isReady: Bool = false
```
Comment (DRadmir) [inline fearless/Modules/NFT/NftDetails/NftDetailsInteractor.swift:9]: can you please explain, why we should know about ready?

File: fearless/Modules/NFT/NftDetails/NftDetailsInteractor.swift [hunk 3]
```diff
@@ -3,22 +3,61 @@ import UIKit
...
+
+    func fetchData() {
+        guard isReady else {
+            return
+        }
+
+        isReady = false
+
+        Task {
+            do {
```
Comment (DRadmir) [inline fearless/Modules/NFT/NftDetails/NftDetailsInteractor.swift:45]: catch the error please

Comment (DRadmir) [reply fearless/Modules/NFT/NftDetails/NftDetailsInteractor.swift:45]: you should change isReady to false if error caught?

File: fearless/Modules/NFT/NftDetails/NftDetailsPresenter.swift [hunk 4]
```diff
@@ -107,6 +117,10 @@ extension NftDetailsPresenter: NftDetailsViewOutput {
         interactor.setup(with: self)
     }
 
+    func viewAppeared() {
+        interactor.initialSetup()
```
Comment (DRadmir) [inline fearless/Modules/NFT/NftDetails/NftDetailsPresenter.swift:121]: it is not initial setup, cause init setup is func setup(with output: NftDetailsInteractorOutput), perhaps viewApeared also?

File: fearless/Modules/NFT/NftDetails/ViewModel/NftDetailViewModelFactory.swift [hunk 5]
```diff
@@ -31,4 +32,12 @@ final class NftDetailViewModelFactory: NftDetailViewModelFactoryProtocol {
             priceString: nil
         )
     }
+
+    func buildOwnerString(owners: [String], address: String, locale _: Locale) -> String {
```
Comment (DRadmir) [inline fearless/Modules/NFT/NftDetails/ViewModel/NftDetailViewModelFactory.swift:36]: locale unused

File: fearless/Modules/NFT/NftDetails/ViewModel/NftDetailViewModelFactory.swift [hunk 6]
```diff
@@ -31,4 +32,12 @@ final class NftDetailViewModelFactory: NftDetailViewModelFactoryProtocol {
...
+        var ownerString = owners.first { ownerAddress in
```
Comment (DRadmir) [inline fearless/Modules/NFT/NftDetails/ViewModel/NftDetailViewModelFactory.swift:37]: looks as let

File: fearless/Modules/NFT/NftDetails/ViewModel/NftDetailViewModelFactory.swift [hunk 7]
```diff
@@ -31,4 +32,12 @@ final class NftDetailViewModelFactory: NftDetailViewModelFactoryProtocol {
...
+            ownerAddress == address
+        } ?? owners.first
+        if owners.count > 1 {}
```
Comment (DRadmir) [inline fearless/Modules/NFT/NftDetails/ViewModel/NftDetailViewModelFactory.swift:40]: unused if

File: fearless/en.lproj/Localizable.strings [hunk 8]
```diff
@@ -1249,3 +1249,4 @@ To find out more, contact %@";
 "vesting.locked.title" = "Vesting Locked";
 "vesting.claim.disclaimer.title" = "Important Notice Regarding Token Claims:";
 "vesting.claim.disclaimer.text" = "Due to the unique vesting schedules of each parachain, our app is unable to display the quantity of locked tokens eligible for claiming. Please be advised that initiating a claim may be impractical if the prospective amount is marginal and comparable to the transaction fee involved. For comprehensive insights into your locked balances, consult the Subscan blockexplorer. We urge you to assess the information carefully and proceed at your own discretion.";
+"common.and.others.placeholder" = "%s & others";
```
Comment (DRadmir) [inline fearless/en.lproj/Localizable.strings:1252]: please don't forget about adding string to localise and also export strings for all locale

**Review by DRadmir — CHANGES_REQUESTED** [review]

File: fearless/CoreLayer/OperationFactory/NFT/AlchemyNFTOperationFactory.swift [hunk 9]
```diff
@@ -332,6 +332,67 @@ final class AlchemyNFTOperationFactory {
...
+
+        let operation = NetworkOperation(
+            requestFactory: requestFactory,
+            resultFactory: resultFactory
+        )
+
+        return operation
+    }
+
+    private func createMapOwnersOperation(
```
[the remark this answers is not in the document]
Comment (Viscount88) [reply fearless/CoreLayer/OperationFactory/NFT/AlchemyNFTOperationFactory.swift:387]: map AlchemyOwnersResponse to owners array.

File: fearless/Modules/NFT/NftDetails/NftDetailsInteractor.swift (hunk 3 above)
Comment (Viscount88) [reply fearless/Modules/NFT/NftDetails/NftDetailsInteractor.swift:45]: There is nothing to catch, if we didn't got owners user shouldn't know about that.

**Review by DRadmir — DISMISSED** [review]

**Review by DRadmir — APPROVED** [review]

**Pull request merged**